src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -490,157 +557,40 @@ private DynamicTraceEventData.PayloadFetchClassInfo CheckForWellKnownEventFields
             return null;
         }
 
-        private DynamicTraceEventData.PayloadFetchClassInfo ParseFields(PinnedStreamReader reader, int numFields)
+        private DynamicTraceEventData.PayloadFetchClassInfo ParseFields(PinnedStreamReader reader, int numFields, EventPipeMetaDataVersion encodingVersion,
```

Comment:
metadataVersion isn't needed (it is only used to pass to ParseType, and ParseType only uses it to pass back to ParseFields)